src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpRequestHeaders.cs

```diff
@@ -159,6 +159,7 @@ private T GetSpecializedCollection<T>(int slot, Func<HttpRequestHeaders, T> crea
             set { SetOrRemoveParsedValue(KnownHeaders.MaxForwards.Descriptor, value); }
         }
 
+        public string? Protocol { get; set; }
```

Comment:
Should any validation happen on this value?

Reply:
Spec says that unknown values should be ignored https://datatracker.ietf.org/doc/html/rfc7540#section-5.5

Reply:
Sure, but what about weird stuff like control characters? Those are filtered out for other headers.

Reply:
It seems that we don't check values for control characters, only headers. In values only newlines are checked https://github.com/dotnet/runtime/blob/32d0360b73bd77256cc9a9314a3c4280a61ea9bc/src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs#L974
Should I add `CheckContainsNewLine`?

Reply:
Added `CheckContainsNewLine`